mlir/lib/TableGen/Pattern.cpp

```diff
@@ -300,6 +340,12 @@ std::string SymbolInfoMap::SymbolInfo::getValueAndRangeUse(
     LLVM_DEBUG(dbgs() << repl << " (Attr)\n");
     return std::string(repl);
   }
+  case Kind::Prop: {
+    assert(index < 0);
```

Comment:
This feels confusing here as you aren't using index. What are you trying to guard against here?